### DISCUSSION TITLE
How can I send additional user context to Sentry to identify the user?

### DISCUSSION THREAD

**Discussion by niazangels** [fastapi/fastapi#9060] [Questions]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I send additional context to Sentry to identify the user who faced the error?

### Additional context

I already use JWT to identify the user, plus I have the lovely dependencies from FastAPI for pulling in additional information from the database such as the user name. I don't know where to leverage this though.

So far, I have
```python
if config.SENTRY_DSN:
    sentry_sdk.init(config.SENTRY_DSN)
    app = SentryAsgiMiddleware(app)
```

which does a fine job of capturing a lot of stuff automatically, but I want to add the user information in the payload. Sentry's website tells me the following is the way to go about:

```python
from sentry_sdk import configure_scope

with configure_scope() as scope:
    scope.user = {"email": "john.doe@example.com"}
```
But I have no idea where to put this- exceptions could happen from any endpoint from any routers. Some of these require the users to be logged in, while some don't. Adding a scope to every router seems like a wrong way to approach this problem. 

What is a good way to go about this? Any suggestions would be much appreciated. 

Thanks in advance.

**Comment by mittalsuraj18** [comment]
@niazangels what you are saying is correct.
you just have to put the configure scope at the root of each api which you want to monitor

something like this

```
@app.get("/test")
def test(request:Request):
    with configure_scope() as scope:
          scope.user = request.user
          ....
          your remaining code
          ....

**Comment by jokull** [comment]
I have a `Depends` like this 

```python
def get_current_user(db: Session = Depends(get_db), token: str = Security(auth)):
    try:
        payload = jwt.decode(token, config("SECRET_KEY"), algorithms=[ALGORITHM])
    except PyJWTError:
        raise HTTPException(
            status_code=HTTP_403_FORBIDDEN, detail="Could not validate credentials"
        )
    user = crud.user.get(db, id=payload["user_id"])
    if not user:
        raise HTTPException(status_code=404, detail="User not found")
    with configure_scope() as scope:
        scope.user = {"id": user.id, "email": user.email}
        try:
            yield user
        except Exception as e:
            capture_exception(e)
            raise
```

used in routes like this

```python
@router.get("/me/_error", response_model=User)
def get_user_error(
    db: Session = Depends(get_db),
    current_user: DBUser = Depends(get_current_active_user),
):
    1 / 0
```

**Comment by lesnik512** [comment]
With `configure_scope` it's not worked for me.

But with `set_user` it's worked:

```python
from sentry_sdk import set_user

async def set_sentry_user(
    request: Request, credentials: Optional[HTTPAuthorizationCredentials] = security
) -> None:
    username = // get username from credentials somehow
    client = request.client
    set_user({"username": username, "ip_address": client.host if client else None})
```

**Reply by Archirk** [reply]
Could anyone please explain how or why you are sure that with configure_scope() it configures for one request and there is no race to overwrite it before it will be sent to sentry?
